### PR TITLE
test(garden-service): add GetTaskResultCommand and GetTestResultCommand tests

### PR DESCRIPTION
**Pull request by benstov**

### CODE REVIEW

**Comment by eysi09** [comment]
We need to merge the `add-info-pane` branch first or change the base branch for this one. Currently it's based on master and also contains the add-info-pane commit (and all its file changes).

**Comment by eysi09** [comment]
Also, we should change the commit scope from `garden-service` to `commands`. So the commit message would start with `test(commands): ...`.

File: garden-service/test/unit/src/commands/get/get-task-result.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { dataDir, makeTestGarden, expectError } from "../../../../helpers"
+import { resolve } from "path"
+import { GetTaskResultCommand } from "../../../../../src/commands/get/get-task-result"
+import { expect } from "chai"
+
+describe("GetTaskResultCommand", () => {
+  it("should not find module by module name", async () => {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: The message doesn't really make sense. Shouldn't it be something like: "should throw error if task not found"

File: garden-service/test/unit/src/commands/get/get-task-result.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const garden = await makeTestGarden(
+      resolve(dataDir, "test-project-dependants"),
+    )
+    const log = garden.log
+    const command = new GetTaskResultCommand()
+
+    await expectError(
+      async () =>
+        await command.action({ garden, log, args: { name }, opts: {} }),
+      "parameter",
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: Not relevant to this PR per se but the command should throw a `NotFound` error if the task name doesn't exist.

File: garden-service/test/unit/src/commands/get/get-task-result.ts (hunk 1 above)
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: The message doesn't really make sense. Shouldn't it be something like: "should throw error if task not found"

File: garden-service/test/unit/src/commands/get/get-task-result.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    )
+  })
+
+  it("should find module", async () => {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: Again, this need a better description. Something like: "it should return the task result".

File: garden-service/test/unit/src/commands/get/get-test-result.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { dataDir, makeTestGarden, expectError } from "../../../../helpers"
+import { resolve } from "path"
+import { GetTestResultCommand } from "../../../../../src/commands/get/get-test-result"
+import { expect } from "chai"
+
+describe("GetTestResultCommand", () => {
+  it("should not find module by module name", async () => {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-test-result.ts]: Same as above for the description.

File: garden-service/test/unit/src/commands/get/get-test-result.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    const garden = await makeTestGarden(
+      resolve(dataDir, "test-project-dependants"),
+    )
+    const log = garden.log
+    const command = new GetTestResultCommand()
+
+    await expectError(
+      async () =>
+        await command.action({ garden, log, args: { name, module }, opts: {} }),
+      "parameter",
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-test-result.ts]: Same as above for the error type.

File: garden-service/test/unit/src/commands/get/get-test-result.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    )
+  })
+
+  it("should find module", async () => {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-test-result.ts]: Same as above for the description.

File: garden-service/test/unit/src/commands/get/get-task-result.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    const command = new GetTaskResultCommand()
+
+    const res = await command.action({
+      garden,
+      log,
+      args: { name },
+      opts: {},
+    })
+
+    expect(res.result).to.exist
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: In general we shouldn't use multiple `expect` statements in the same `it` function. If there are multiple cases they should be handled my multiple `it`s.

File: garden-service/test/unit/src/commands/get/get-task-result.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (res.result) {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-task-result.ts]: Again with the multiple `expect`s. Here you could just do something like:

```javascript
expect(res.result).to.eql({ output: null, name })
```

Notice also the difference between `equal` and `eql`. The former is for strictly equals (`===`) and the latter for deep objects equals. (IMO it's a terrible convention. There's also `deepEqual` which is the same as `eql` but way clearer)`

File: garden-service/test/unit/src/commands/get/get-test-result.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    const command = new GetTestResultCommand()
+
+    const res = await command.action({
+      garden,
+      log,
+      args: { name, module },
+      opts: {},
+    })
+
+    expect(res.result).to.exist
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-test-result.ts]: Same as above.

File: garden-service/test/unit/src/commands/get/get-test-result.ts [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+    if (res.result) {
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/get/get-test-result.ts]: Same as above. 

The reason that it's better to have a single `expect` per `it` is that it's easier to debug. If one of the fails you don't know which one.

**Comment by eysi09** [comment]
@benstov Would be good to resolve the comments and get this one approved.

**Comment by eysi09** [comment]
@benstov bump. Please resolve the comments so that we can close this one.

File: garden-service/test/unit/src/commands/get/get-task-result.ts (hunk 8 above)
Comment (benstov) [reply garden-service/test/unit/src/commands/get/get-task-result.ts]: thanks, this is really helpful!

**Pull request merged**